pgmpy/models/DynamicBayesianNetwork.py

```diff
@@ -0,0 +1,481 @@
+        >>> dbn.add_edges_from([(('D',0),('G',0)),(('I',0),('G',0)),(('G',0),('L',0))])
+
+        most of the methods will be imported from Bayesian Model
+        However there will be a few changes
+        >>> dbn.list_of_nodes
+        ['D','G','I','S','L']
+        >>> dbn.edges()
+        [(('D',0),('G',0)),(('I',0),('G',0)),(('G',0),('L',0))]
+        If some edges connect nodes not yet in the model, the nodes
+        are added automatically.  There are no errors when adding
+        nodes or edges that already exist.
+
+        Methods:
+        -------
+        add_nodes_from
+        add_edge
+        add_edges_from
+        intra_slice
+        inter_slice
+        add_cpds
+        get_cpds
+        compute_initial_state
+        unroll
+        remove_timeslice
+        """
+        super().__init__()
+        self.inter_edges = []
+        self.intra_edges = []
+        self.dbn_cpds = OrderedDict()
+        self.list_of_nodes = []
+        if ebunch:
+            self.add_nodes_from(ebunch)
+        self.timestate = 1
+        self.computed = False
+
+    def add_node(self, node):
+        """
+        adding node in the static Bayesian Network.
+
+        Parameters:
+        ----------
+        nodes: nodes
+               Nodes can be any hashable python object.
+
+        EXAMPLE
+        --------
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> dbn = DBN()
+        >>> dbn.add_node('D')
+        """
+        self.list_of_nodes.extend(node)
+
+    def add_edge(self, start, end, **kwargs):
+        """
+        Add an edge between two nodes.
+
+        The nodes will be automatically added if they are
+        not already in the initial static Bayesian Network.
+
+        Parameters
+        ----------
+        start, end: The start, end nodes should contain the
+                    (node_name, time_slice)
+                    Here, node_name can be a hashable python object
+                    while the time_slice is an integer value, which
+                    denotes the index of the time_slice that the node
+                    belongs to.
+
+        EXAMPLE
+        -------
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> dbn = DBN()
+        >>> dbn.add_nodes_from(['d', 'i'])
+        >>> dbn.add_edge((('d',0), ('i',0)))
+        """
+        edge = (start, end)
+        if not any(map(lambda x: isinstance(x, tuple), iter(edge))):
+            raise AttributeError('The edge is not of a standard form')
+
+        if not any(map(lambda x: isinstance(x[1], int), iter(edge))):
+            raise AttributeError('The edge does not contain the timeslice in integer form')
+
+        if not any(map(lambda x: x[0] in self.list_of_nodes, iter(edge))):
+            new_nodes = list(filter(lambda x: x[0] not in self.list_of_nodes, iter(edge)))
+            self.add_nodes_from(set(map(lambda x: x[0], iter(new_nodes))))
```

Comment:
@palashahuja Yes remove the `list_of_nodes` variable. Store the node names in `self.nodes` and add a method that overrides the netowrkx's `nodes()` method and it will return just the name of the variables `['D', 'G', 'I', 'L']`.